pkg/acsengine/engine.go

```diff
@@ -873,8 +873,14 @@ func (t *TemplateGenerator) getTemplateFuncMap(cs *api.ContainerService) templat
 			}
 			return buf.String()
 		},
-		"GetKubeletConfigKeyVals": func() string {
+		"GetKubeletConfigKeyVals": func(profile *api.AgentPoolProfile) string {
```

Comment:
I think we should change the function arg here to a `*api.KubernetesConfig`, and just make this a simple string convenience function (look for a `KubeletConfig` of type `map[string]string`, and then concatenate all the key/vals together).

We then move all the "what type of KubernetesConfig are you?" logic to the caller, and not inside the function.

What am I missing that would suggest we should do it this way instead?